swap_meet/vendor.py

```diff
@@ -0,0 +1,207 @@
+#_______________WAVE 1_______________________
+''' from item import Item
+from clothing import Clothing
+from decor import Decor
+from electronics import Electronics '''
+
+class Vendor:
+    def __init__(self, inventory = None): 
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+     
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        for thing in self.inventory:
+            if item == thing:
+                self.inventory.remove(thing)
+                return item
+            
+        return False         
+        
+#_______________WAVE 2_________________________
+    #every instance of the item class becomes an element in the inventory list
+    #returns list of class Item's objects inventory = [Item_a, Item_b, Item_c]
+    
+    def get_by_category(self, category):
+        new_list_with_items = []
+        
+        #iterate on the list for each category (state) and check if the item is in there 
+        for thing in self.inventory:
+            if thing.category == category:
+                new_list_with_items.append(thing)
+    
+        return new_list_with_items
+
+#_______________Wave 3___________________________
+   
+    #helper function
+    def does_item_exist(self, item, item_list):
+        for i in item_list:
+            if i == item:
+                return True
+        return False
+            
+    def swap_items(self, Vendor, my_item, their_item):
+        my_inventory = self.inventory
+        their_inventory = Vendor.inventory
```

Comment:
`Vendor` should be a variable like `other_vendor` here. It ends up being ok, because we essentially overwrite the `Vendor` class with whatever's passed in here in the method scope, but it would be more clear if we used a variable name like `other_vendor`